pkg/s3store/s3store.go

```diff
@@ -100,6 +101,91 @@ import (
 // considered valid into a header value according to RFC2616.
 var nonPrintableRegexp = regexp.MustCompile(`[^\x09\x20-\x7E]`)
 
+var _ handler.ServerDataStore = &S3Store{}
+
+func (store S3Store) AsServableUpload(upload handler.Upload) handler.ServableUpload {
+	return &S3ServableUpload{
```

Comment:
If we implement `ServeContent` directly on `s3Upload`, we do not have to introduce another struct `S3ServableUpload`. This is already what we do for `AsTerminatableUpload`:

https://github.com/tus/tusd/blob/9779a84cae29d083b3496ba732fe7c76755bd4ed/pkg/s3store/s3store.go#L367-L369

Reply:
I can change this, though I did see value in it.